hail/src/main/scala/is/hail/expr/ir/GenericLines.scala

```diff
@@ -298,6 +309,85 @@ object GenericLines {
     GenericLines.read(fs, contexts, gzAsBGZ, filePerPartition)
   }
 
+  def readTabix(fs: FS, fileStatus: FileStatus, contigMapping: Map[String, String], partitions: IndexedSeq[Interval]): GenericLines = {
+    val reverseContigMapping: Map[String, String] =
+      contigMapping.toArray
+        .groupBy(_._2)
+        .map { case (target, mappings) =>
+          if (mappings.length > 1)
+            fatal(s"contig_recoding may not map multiple contigs to the same target contig, " +
+              s"due to ambiguity when querying the tabix index." +
+            s"\n  Duplicate mappings: ${ mappings.map(_._1).mkString(",") } all map to ${ target }")
+            (target, mappings.head._1)
+        }.toMap
+    val contexts = partitions.zipWithIndex.map { case (interval, i) =>
+      val start = interval.start.asInstanceOf[Row].getAs[Locus](0)
+      val end = interval.end.asInstanceOf[Row].getAs[Locus](0)
+      val contig = reverseContigMapping.getOrElse(start.contig, start.contig)
+      Row(i, fileStatus.getPath, contig, start.position, end.position)
+    }
+    val body: (FS, Any) => CloseableIterator[GenericLine] = { (fs: FS, context: Any) =>
+      val contextRow = context.asInstanceOf[Row]
+      val index = contextRow.getAs[Int](0)
+      val file = contextRow.getAs[String](1)
+      val chrom = contextRow.getAs[String](2)
+      val start = contextRow.getAs[Int](3)
+      val end = contextRow.getAs[Int](4)
+
+      val reg = {
+        val r = new TabixReader(file, fs)
+        val tid = r.chr2tid(chrom)
+        r.queryPairs(tid, start - 1, end)
```

Comment:
Is this robust to `start` being the first position? i.e. passing `0` to queryPairs is fine?

Reply:
Yes. We do so all the time in the combiner with default genome intervals.